corehq/apps/app_manager/suite_xml/post_process/remote_requests.py

```diff
@@ -338,7 +338,8 @@ def build_query_prompts(self):
             if prop.validations:
                 kwargs['validations'] = [
                     Validation(
-                        test=interpolate_xpath(validation.test),
+                        # don't interpolate dots since "." is a valid reference here
```

Comment:
Can you explain this further?  Does this mean that `.` is fine - like we can send it to mobile and it will work as expected?  Or that we intend to support `.` in this context, perhaps by replacing it with a case reference?

I'm wondering if the fix should instead be to pass in a valid `case_xpath` to this call.

Reply:
I see the difficulty now.  That's tricky.  I'm tempted to just drop the dot pattern from the validation conditions - if they put a dot where it's not valid, they'll get an error at run time instead of at build time, and we won't have HQ attempt to override those dots unless a `case_xpath` is provided.  From the comments, looks like this code was originally written only for module and form filters (contexts where it's reasonable to assume there's a case), but in this context, things are a bit different.  Your change does seem safer.

Reply:
I think it makes sense to have the build validation when case xpath is needed, because it could be a useful hint on how to fix the error (as long as the messaging is accurate). This seems to be a somewhat unique circumstance where a reference that would _usually_ require case xpath does not. There are also about 9 other uses of `interpolate_xpath` outside of tests, and I can't confidently say that it'd be safe for all of them to remove the dot pattern validation.

Digging a little deeper, I think part of the difficulty is that `interpolate_xpath` is being used for both interpolation and validation, but it's not clear which calls need which functionality from it. For example, is it appropriate to throw an error if someone tries to use `#case` in a place where it will never get `case_xpath` passed in? Or should it be allowed through because that reference couldn't ever work and (probably) won't cause harm?

Reply:
Yeah, its usage feels disjointed - different contexts, different use-cases.  There are three ways dots are treated too:
* fail, because we don't have a case reference to replace them with
* replace them with a case reference
* leave them be

I was thinking about ways to present an interface that makes this behavior clearer (maybe do dot interpolation as a separate function call, not part of `interpolate_xpath`?), but couldn't really come up with anything better than a flag on the main function like you've done here.